myhpom/storage.py

```diff
@@ -0,0 +1,22 @@
+from django.core.files.storage import Storage
+from django.core.urlresolvers import reverse
+
+from django_irods.storage import IrodsStorage
+
+
+class MyhpomStorage(IrodsStorage):
+    """
+    An IRODs based storage method that assigns filenames the same way that the
+    default Storage class does, and uses MYHPOM specific URLs to access them.
+    """
+
+    def url(self, name):
+        # Rather than relying on views in the irods package (which require yet
+        # other hydroshare specific applications), provide one for myhpom:
+        return reverse('myhpom:irods_download', kwargs={'path': name})
+
+    def get_available_name(self, name):
+        # Use the default method of creating the name used to save 'name' as.
+        # The IrodsStorage implementation will prevent files with the same
+        # 'name' from being stored:
+        return Storage.get_available_name(self, name)
```

Comment:
On one hand, if we do the fix below and there are never files left lying around, then the security hole is never hit. 

But I think I would be most comfortable with the second option — always append a random hash or at least short string to the basename ... at least for users' advance directives?

Reply:
Okay, I'll do that - I realize part of #1 will have to happen anyway...because eventually Gerald *will* notice there are filename clashes for state advance directives. I guess I'll make a task in the backlog for that.

Reply:
Created MH-177 for ^